FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/report_builder.c

```diff
@@ -323,6 +379,185 @@ static eReportBuilderStatus prvWriteConnectionsArray( char * pcBuffer,
 }
 /*-----------------------------------------------------------*/
 
+static eReportBuilderStatus prvWriteCustomMetrics( char * pcBuffer,
+                                                   uint32_t ulBufferLength,
+                                                   const CustomMetric_t * pxCustomMetricsArray,
+                                                   uint32_t ulCustomMetricsArrayLength,
+                                                   uint32_t * pulOutCharsWritten )
+{
+    char * pcCurrentWritePos = pcBuffer;
```

Comment:
This function is longer than ideal, it can be made more readable by having utility functions for writing JSON for each of the custom metric types (like `prvWriteNumberMetric`, `prvWriteStringListMetric`, etc). This function can then call the specific metric type's utility function instead of having all the code in the single function.

Reply:
Splitting the function that way would be worse as the part that depends on the type is short. The surrounding code would be duplicated, which would be harder to maintain and harder to understand as the functions would be the same with minor differences. Also, the code to call the specific functions would be similar to the current code, thus not gaining us much.